### DISCUSSION TITLE
stream lagging a lot on remote server.

### DISCUSSION THREAD

**Discussion by Anurag-Raut** [bluenviron/mediamtx#2052] [Questions]
when i am running the mediamtx server on local machine it is working fine.
to run on cloud server ,I am using digital ocean droplet linux server ,  I have setup a domain name, and then using reverse proxy using nginx like this-

```
server {
    listen 8886 ssl;
    server_name example1.com;

    ssl_certificate /etc/letsencrypt/live/example1.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/example1.com/privkey.pem;

    location / {
        proxy_pass http://localhost:8887;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

server {
    listen 8887 ssl;
    server_name example2.com;

    ssl_certificate /etc/letsencrypt/live/example2.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/example2.com/privkey.pem;

    location / {
        proxy_pass http://localhost:8889;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}


```

But running with this configuration results into very laggy and blurry stream with green patches in between.
like this - 
![image](https://github.com/bluenviron/mediamtx/assets/102049482/41969ab1-f251-47d9-9ea2-60f6f3aefa5d)

but it works perfect on loaclhost.
how to fix this , or is there another reliable way to run mediamtx server remotely other than aws.
I would be glad if anyone could help .

Thanks

**Comment by aler9** [comment]
Hello, in my opinion corruption doesn't happen when the stream is read with HLS, since HLS is a really robust protocol with respect to corruption, but it happens when the stream is sent to the server, probably with RTSP or RTMP.

Take a look at the README section about it: https://github.com/bluenviron/mediamtx#corrupted-frames

[a comment by an automated account is not shown]
